homeassistant/components/saj/sensor.py

```diff
@@ -140,8 +141,8 @@ def stop_update_interval(event):
         """Properly cancel the scheduled update."""
         remove_interval_update()  # pylint: disable=not-callable
 
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, start_update_interval)
     hass.bus.async_listen(EVENT_HOMEASSISTANT_STOP, stop_update_interval)
+    async_at_start(hass, start_update_interval)
```

Comment:
We need to decorate `start_update_interval` with `@callback` imported from core.py.

Reply:
Thanks for your review. Do i only fix the start_update_interview or should i add it to the stop function too.

Reply:
Please add it to both. Thanks!